spec/system/admin/reports/orders_and_distributors_spec.rb

```diff
@@ -0,0 +1,179 @@
+# frozen_string_literal: true
+
+require "system_helper"
+
+RSpec.describe "Orders And Distributors" do
+  include AuthenticationHelper
+  include WebHelper
+  include ReportsHelper
+
+  describe "Orders And Distributors" do
+    let!(:distributor) { create(:distributor_enterprise, name: "By Bike") }
+    let!(:distributor2) { create(:distributor_enterprise, name: "By Moto") }
+    let!(:completed_at) { Time.zone.now.to_fs(:db) }
+
+    let!(:order) {
+      create(:order_ready_to_ship, distributor_id: distributor.id, completed_at:)
+    }
+    let!(:order2) {
+      create(:order_ready_to_ship, distributor_id: distributor2.id, completed_at:)
+    }
+
+    context "as an enterprise user" do
+      let(:header) {
+        ["Order date", "Order Id", "Customer Name", "Customer Email", "Customer Phone",
+         "Customer City", "SKU", "Item name", "Variant", "Quantity", "Max Quantity",
+         "Cost", "Shipping Cost", "Payment Method", "Distributor", "Distributor address",
+         "Distributor city", "Distributor postcode", "Shipping Method",
+         "Shipping instructions"]
+      }
+      let(:line_item1) {
+        [completed_at, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item2) {
+        [completed_at, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item3) {
+        [completed_at.to_s, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item4) {
+        [completed_at.to_s, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item5) {
+        [completed_at.to_s, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+
+      before do
+        login_as(distributor.owner)
+        visit admin_reports_path
+        click_link "Orders And Distributors"
+        run_report
+      end
+
+      it "generates the report" do
+        rows = find("table.report__table").all("thead tr")
+        table_headers = rows.map { |r| r.all("th").map { |c| c.text.strip } }
```

Comment:
We don't have a helper for this?

Reply:
Hum, true this is used throughout report specs, although with slight variations. Would be great to unify this in a helper.